spring-ai-spring-boot-testcontainers/src/test/java/org/springframework/ai/testcontainers/service/connection/opensearch/AwsOpenSearchContainerConnectionDetailsFactoryTest.java

```diff
@@ -0,0 +1,147 @@
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.DefaultResourceLoader;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.hasSize;
+
+@SpringJUnitConfig
+@TestPropertySource(properties = { "spring.ai.vectorstore.opensearch.index-name=auto-spring-ai-document-index",
+		"spring.ai.vectorstore.opensearch.initialize-schema=true",
+		"spring.ai.vectorstore.opensearch.mapping-json="
+				+ AwsOpenSearchContainerConnectionDetailsFactoryTest.MAPPING_JSON,
+		"spring.ai.vectorstore.opensearch.aws.domain-name=testcontainers-domain",
+		"spring.ai.vectorstore.opensearch.aws.service-name=es" })
+@Testcontainers
+class AwsOpenSearchContainerConnectionDetailsFactoryTest {
+
+	static final String MAPPING_JSON = "{\"properties\":{\"embedding\":{\"type\":\"knn_vector\",\"dimension\":384}}}";
+
+	@Container
+	@ServiceConnection
+	private static final LocalStackContainer localstack = new LocalStackContainer(
+			DockerImageName.parse("localstack/localstack:3.5.0"))
+		.withEnv("LOCALSTACK_HOST", "localhost.localstack.cloud");
+
+	private final List<Document> documents = List.of(
+			new Document("1", getText("classpath:/test/data/spring.ai.txt"), Map.of("meta1", "meta1")),
+			new Document("2", getText("classpath:/test/data/time.shelter.txt"), Map.of()),
+			new Document("3", getText("classpath:/test/data/great.depression.txt"), Map.of("meta2", "meta2")));
+
+	@Autowired
+	private VectorStore vectorStore;
+
+	@BeforeAll
+	static void beforeAll() throws IOException, InterruptedException {
+		String[] createDomainCmd = { "awslocal", "opensearch", "create-domain", "--domain-name",
+				"testcontainers-domain", "--region", localstack.getRegion() };
+		localstack.execInContainer(createDomainCmd);
+
+		String[] describeDomainCmd = { "awslocal", "opensearch", "describe-domain", "--domain-name",
+				"testcontainers-domain", "--region", localstack.getRegion() };
+		await().pollInterval(Duration.ofSeconds(30)).atMost(Duration.ofSeconds(300)).untilAsserted(() -> {
+			org.testcontainers.containers.Container.ExecResult execResult = localstack
+				.execInContainer(describeDomainCmd);
+			String response = execResult.getStdout();
+			JSONArray processed = JsonPath.read(response, "$.DomainStatus[?(@.Processing == false)]");
```

Comment:
@eddumelendez Thanks for the PR. This test consistently times out here. I believe this is probably an environment issue but want to check with you.

Reply:
@eddumelendez suggested to increase the timeout value as the localstack tries to pull the open search binaries could take longer depending on the internet bandwidth. It worked after increasing to a longer timeout on my local. Thanks!